### PR TITLE
Fixes for changes in marketplace-dev nav

### PR DESCRIPTION
**Pull request by bobsilverberg**
This fixes a large number of failing tests on dev and stage, which are failing due to a new nav being implemented.

Specifically, this should allow all the tests in `tests/desktop/consumer_pages/test_home_page.py` and `tests/desktop/consumer_pages/test_details_page.py` to pass.

I will put together another PR for the other failing tests.

@krupa We need to decide whether to merge these changes to the `master` branch, in which case prod will fail until this new nav is implemented on prod, or whether we need to have a new branch to use for just testing dev and stage. How long will it be before the changes make it to prod?

@m8ttyB / @rbillings r?

### CODE REVIEW

**Comment by bobsilverberg** [comment]
I have updated this PR and now all tests should be passing, although two review tests still have intermittent issues as they are awaiting the merge of PR#648.

File: tests/desktop/consumer_pages/test_home_page.py [hunk 1]
```diff
@@ -34,7 +34,7 @@ def test_that_header_has_expected_items(self, mozwebqa):
 
         Assert.true(home_page.header.is_logo_visible)
         Assert.true(home_page.header.is_search_visible)
-        Assert.equal(home_page.header.search_field_placeholder, "Search the Marketplace")
+        Assert.equal(home_page.header.search_field_placeholder, u'Search Marketplace\u2026')
```
Comment (m8ttyB) [inline tests/desktop/consumer_pages/test_home_page.py]: Nit, can you add a comment pertaining to the importance of this unicode character?

Comment (m8ttyB) [inline tests/desktop/consumer_pages/test_home_page.py]: Actually, nm.

**Comment by m8ttyB** [comment]
r+ and merging

**Pull request merged**